torch/distributed/fsdp/_init_utils.py

```diff
@@ -817,6 +817,16 @@ def _get_device_from_device_id(
             "index as the `device_id` argument."
         )
         device = torch.device("cuda", torch.cuda.current_device())
+    if device == torch.device("hpu"):
```

Comment:
nit: it would be nicer to avoid the copy-paste here.  Could the code be written in a way to have a list of device types and their cooresponding current_device functions, then just iterate this list and issue the warning/return inside the loop?

Reply:
done